portfolio/src/main/webapp/script.js

```diff
@@ -32,53 +32,59 @@ function addRandomGreeting() {
-  let numCommentsSelect = document.getElementById('num-comments');
-  let numCommentsString = numCommentsSelect.options[numCommentsSelect.selectedIndex].value;
-
-  // Create fetch string and perform GET request.
-  let searchParams = new URLSearchParams();
-  searchParams.append('num-comments', encodeURIComponent(numCommentsString));
-  fetch('/data?' + searchParams).then(response => response.json()).then((comments) => {
-    let commentsContainer = document.getElementById('comments-container');
-
-    // Reset comments continer.
-    commentsContainer.innerHTML = '';
-
-    comments.forEach((comment) => {
-      // Create HTML for comment.
-      let commentP = document.createElement("p");
-      commentP.innerHTML = comment;
-
-      // Add comment paragraph to comments-box div
-      let commentDiv = document.createElement('div');
-      commentDiv.classList.add('comments-box');
-      commentDiv.appendChild(commentP);
-
-      // Add comment div to container.
-      commentsContainer.appendChild(commentDiv);
+  if (getUserLogin()) {
+    // Extract the number of comments selected from the dropdown.
+    let numCommentsSelect = document.getElementById('num-comments');
+    let numCommentsString = numCommentsSelect.options[numCommentsSelect.selectedIndex].value;
+
+    // Create fetch string and perform GET request.
+    let searchParams = new URLSearchParams();
+    searchParams.append('num-comments', encodeURIComponent(numCommentsString));
+    fetch('/data?' + searchParams).then(response => response.json()).then((comments) => {
+      let commentsContainer = document.getElementById('comments-container');
+
+      // Reset comments continer.
+      commentsContainer.innerHTML = '';
+
+      comments.forEach((comment) => {
+        // Create HTML for comment.
+        let commentP = document.createElement("p");
+        commentP.innerHTML = comment;
+
+        // Add comment paragraph to comments-box div
+        let commentDiv = document.createElement('div');
+        commentDiv.classList.add('comments-box');
+        commentDiv.appendChild(commentP);
+
+        // Add comment div to container.
+        commentsContainer.appendChild(commentDiv);
+      });
     });
-  });
+  }
 }
 
 /**
  * Sends request to delete all comments if user confirms that that is the action
  * they want to take.
  */
 function deleteComments() {
-  let doDeleteComments = confirm('Are you sure you want to delete all of the comments?');
+  if (getUserLogin()) {
+    let doDeleteComments = confirm('Are you sure you want to delete all of the comments?');
 
-  if (doDeleteComments === true) {
-    fetch('/delete-data', {
-      method:'POST'
-    }).then(response => response.json()).then(_ => getComments());
+    if (doDeleteComments === true) {
+      fetch('/delete-data', {
+        method:'POST'
+      }).then(response => response.json()).then(_ => getComments());
+    }
+  } else {
+    alert("You must be logged in to delete comments.")
```

Comment:
While it's good that you're handling the UI experience here, remember: The REAL check needs to happen in the servlet.

If someone can just open the javascript console and do `document.getElementById("comment-form").submit();` or post a delete request or something, then that's the real hole. (disregard this if you're already checking this in the comment servlet or if you have another PR open with that change already).

Reply:
You make a good point. I realized this hole, but I wasn't sure how to go about handling it. I will be sure to check for this in the servlet as well.

Reply:
I changed it.